scripts/bootstrap.js

```diff
@@ -0,0 +1,149 @@
+    pre.map(key => tasks[key]).forEach(task => {
+      if (!task.check()) {
+        task.command();
+      }
+    });
+
+    log.info(prefix, name);
+    command();
+  },
+});
+
+const tasks = {
+  reset: createTask({
+    name: `Clean and re-install root dependencies ${chalk.red('(reset)')}`,
+    defaultValue: false,
+    option: '--reset',
+    command: () => {
+      log.info(prefix, 'git clean');
+      spawn('git clean -fdx');
+      log.info(prefix, 'yarn install');
+      spawn('yarn install --no-lockfile');
+    },
+  }),
+  core: createTask({
+    name: `Core & Examples ${chalk.gray('(core)')}`,
+    defaultValue: true,
+    option: '--core',
+    command: () => {
+      spawn('yarn bootstrap:core');
+    },
+  }),
+  docs: createTask({
+    name: `Documentation ${chalk.gray('(docs)')}`,
+    defaultValue: false,
+    option: '--docs',
+    command: () => {
+      spawn('yarn bootstrap:docs');
+    },
+  }),
+  packs: createTask({
+    name: `Build tarballs of packages ${chalk.gray('(build-packs)')}`,
+    defaultValue: false,
+    option: '--packs',
+    command: () => {
+      spawn('yarn build-packs');
+    },
+    check: () => getDirectories(join(__dirname, '..', 'packs')).length > 0,
+  }),
+  'react-native-vanilla': createTask({
+    name: `React-Native example ${chalk.gray('(react-native-vanilla)')}`,
+    defaultValue: false,
+    option: '--reactnative',
+    pre: ['packs'],
+    command: () => {
+      spawn('yarn bootstrap:react-native-vanilla');
+    },
+  }),
+  'crna-kitchen-sink': createTask({
+    name: `React-Native-App example ${chalk.gray('(crna-kitchen-sink)')}`,
+    defaultValue: false,
+    option: '--reactnativeapp',
+    pre: ['packs'],
+    command: () => {
+      spawn('yarn bootstrap:crna-kitchen-sink');
+    },
+  }),
+};
+
+Object.keys(tasks)
+  .reduce((acc, key) => acc.option(tasks[key].option, tasks[key].name), main)
+  .parse(process.argv);
+
+Object.keys(tasks).forEach(key => {
+  tasks[key].value = program[tasks[key].option.replace('--', '')] || program.all;
+});
+
+let selection;
+if (!Object.keys(tasks).map(key => tasks[key].value).filter(Boolean).length) {
+  selection = inquirer
+    .prompt([
+      {
+        type: 'checkbox',
+        message: 'Select which packages to bootstrap',
+        name: 'todo',
+        choices: Object.keys(tasks).map(key => ({
+          name: tasks[key].name,
+          checked: tasks[key].defaultValue,
+        })),
+      },
+    ])
+    .then(answers =>
+      answers.todo.map(name => tasks[Object.keys(tasks).find(i => tasks[i].name === name)])
+    );
+} else {
+  selection = Promise.resolve(
+    Object.keys(tasks).map(key => tasks[key]).filter(item => item.value === true)
+  );
+}
+
+selection.then(list => {
+  if (list.length === 0) {
+    log.warn(prefix, 'Nothing to bootstrap');
+  } else {
+    list.forEach(key => {
+      key.command();
+    });
+    process.stdout.write('\x07');
+    try {
+      spawn('say "Bootstrapping sequence complete"');
```

Comment:
Lol, I'm actually concerned this will be annoying to linux and windows users, will have to ask @UsulPro maybe?